### PR TITLE
Convert ViewAllergy.test.tsx to RTL

### PR DESCRIPTION
**Pull request by Proful**

### CODE REVIEW

File: src/__tests__/patients/allergies/ViewAllergy.test.tsx [hunk 1]
```diff
@@ -19,28 +19,23 @@ describe('View Care Plan', () => {
     jest.spyOn(PatientRepository, 'find').mockResolvedValue(patient)
     const history = createMemoryHistory()
     history.push(`/patients/${patient.id}/allergies/${patient.allergies![0].id}`)
-    let wrapper: any
 
     await act(async () => {
```
Comment (Proful) [inline src/__tests__/patients/allergies/ViewAllergy.test.tsx:22]: Need some suggestions why the await act is needed?

File: src/__tests__/patients/allergies/ViewAllergy.test.tsx [hunk 2]
```diff
@@ -19,28 +19,23 @@ describe('View Care Plan', () => {
...
-      wrapper = await mount(
+      await render(
```
Comment (Proful) [inline src/__tests__/patients/allergies/ViewAllergy.test.tsx:23]: why await (async) render is needed in this case?

Comment (nobrayner) [reply src/__tests__/patients/allergies/ViewAllergy.test.tsx:23]: I don't think this is needed... You can try pulling the render call outside of the act, remove the act, and see if test still passes?

Comment (JacobMGEvans) [reply src/__tests__/patients/allergies/ViewAllergy.test.tsx:23]: Yeah this is likely someone trying to get rid of act warnings in the console lol

Comment (Proful) [reply src/__tests__/patients/allergies/ViewAllergy.test.tsx:23]: Nope! test is failing.. (after removal of act & await)

TestingLibraryElementError: Unable to find an element with the display value: /some name/.

The `input` node is not getting rendered. 

----
Test failing without act.
Tried with both await render & render (without act)

Comment (nobrayner) [reply src/__tests__/patients/allergies/ViewAllergy.test.tsx:23]: Hmm... I would then try leaving out the act still, but adding a waitFor(() => expect(...)) in the actual test body. @JacobMGEvans ?

Comment (Proful) [reply src/__tests__/patients/allergies/ViewAllergy.test.tsx:23]: Tried removing act & await with render and added waitFor surrounding assertions. Test seems to passed.

However two warning
```
(node:13110) UnhandledPromiseRejectionWarning: TypeError: MutationObserver is not a constructor
(node:13110) UnhandledPromiseRejectionWarning: Unhandled promise rejection
```

Comment (JacobMGEvans) [reply src/__tests__/patients/allergies/ViewAllergy.test.tsx:23]: We can come up behind later and deal with cleaning up lol

**Pull request merged**